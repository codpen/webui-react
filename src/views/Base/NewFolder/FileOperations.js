import React from 'react';
import {Button} from "reactstrap";
import NewFolder from "./NewFolder";
import PropTypes from "prop-types";
import Input from "reactstrap/es/Input";
import {connect} from "react-redux";
import {changeGridMode, changeVisibilityFilter} from "../../../actions/explorerStateActions";

class FileOperations extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            newFolderModalIsVisible: false
        };
        this.filterOptions = ["Images", "Pdf", "Videos"];
    }

    openNewFolderModal = () => {
        this.setState({newFolderModalIsVisible: true})
    };

    closeNewFolderModal = () => {
        this.setState({newFolderModalIsVisible: false});
        // this.props.updateHandler();
    };

    handleChangeFilter = (e) => {
        const newFilter = e.target.value;

        this.props.changeVisibilityFilter(this.props.containerID, newFilter);

    };


    handleChangeGridMode = (e) => {
        const gridMode = e.target.value;

        this.props.changeGridMode(this.props.containerID, gridMode);

    };


    render() {

        return (

            <div className="float-right mb-3 mt-1 form-inline">
                <Button color="success" className="mr-1" onClick={this.openNewFolderModal}>New Folder</Button>
                {/*<Button color="success" className="ml-2">Copy from URL</Button>*/}
                {/*<Button color="success" className="ml-2">Copy</Button>*/}
                {/*<Button color="success" className="ml-2">Rename</Button>*/}
                {/*<Button color="success" className="ml-2">Upload</Button>*/}
                <NewFolder containerID={this.props.containerID} isVisible={this.state.newFolderModalIsVisible}
                           closeModal={this.closeNewFolderModal}/>


                <Input type={"select"} onChange={this.handleChangeFilter} value={this.props.visibilityFilter}
                       className="ml-1 mr-1">
                    <option key={0}>None</option>
                    {
                        this.filterOptions.map((item, idx) => {
                            return (<option key={item} value={item}>{item}</option>)
                        })
                    }
                </Input>
                <Input type={"select"} onChange={this.handleChangeGridMode} value={this.props.gridMode}>
                    <option value={"grid"}>Grid</option>
                    <option value={"card"}>Card</option>
                </Input>

            </div>


        );
    }
}

FileOperations.propTypes = {
    updateHandler: PropTypes.func.isRequired,
    containerID: PropTypes.string.isRequired,
    changeVisibilityFilter: PropTypes.func.isRequired,
    visibilityFilter: PropTypes.string,
    gridMode: PropTypes.string
};

const mapStateToProps = (state, ownProps) => ({
    visibilityFilter: state.explorer.visibilityFilters[ownProps.containerID],
    gridMode: state.explorer.gridMode[ownProps.containerID]


});


export default connect(mapStateToProps, {changeVisibilityFilter, changeGridMode})(FileOperations);