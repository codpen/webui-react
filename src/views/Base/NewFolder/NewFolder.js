import React from 'react';
import {Button, Col, Form, FormGroup, Input, Label, Modal, ModalBody, ModalFooter, ModalHeader} from "reactstrap";
import PropTypes from "prop-types";
import axiosInstance from "../../../utils/API/API";
import {toast} from "react-toastify";
import {addColonAtLast} from "../../../utils/Tools";
import {connect} from "react-redux";
import {getFilesForContainerID} from "../../../actions/explorerStateActions";


class NewFolder extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            name: "",
            disableForm: false
        };
        this.createNewFolder = this.createNewFolder.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
        this.toggle = this.toggle.bind(this);

    }

    disableForm = (shouldDisable) => {
        this.setState({disableForm: shouldDisable});
    };

    async createNewFolder() {

        // console.log("Form Submitted");
        let {name} = this.state;
        let {remoteName, remotePath} = this.props.currentPath;
        const {fsInfo} = this.props;

        remoteName = addColonAtLast(remoteName);

        // remotePath = this.state.name;
        try {
            console.log("fsInfo", fsInfo);
            if (fsInfo.Features.BucketBased && remotePath === "") {/*Trying to create a bucket, not a dir*/
                remoteName += name;
            } else { /*Normal directory*/
                if (remotePath === "") {
                    remotePath = name;
                } else {
                    remotePath += "/" + name;
                }
            }
            const data = {
                fs: remoteName,
                remote: remotePath
            };

            /*Disable form submit button*/
            this.disableForm(true);

            /*Network Request*/
            await axiosInstance.post("operations/mkdir", data);

            this.disableForm(false);

            this.toggle();
            toast.info(`Folder created: ${remotePath}`)
            this.props.getFilesForContainerID(this.props.containerID);
        } catch (error) {
            this.disableForm(false);

            if (error.response) {
                toast.error(`Error creating folder: ${error.response.data.error}`)
            } else {
                toast.error(`Error creating folder: ${error}`);
            }

            // console.log(`Error occurred at operations/mkdir: ${e}, ${e.response}`);
        }
    }

    handleSubmit(e) {
        e.preventDefault();
        this.createNewFolder();
    }

    changeName = e => {
        const value = e.target.value;
        this.setState(
            {name: value}
        );
    };

    toggle() {
        this.props.closeModal();
    }

    render() {
        const {name, disableForm} = this.state;
        const {isVisible} = this.props;
        return (

            <Modal isOpen={isVisible} toggle={this.toggle} data-test="parent">
                <Form onSubmit={this.handleSubmit}>
                    <ModalHeader toggle={this.toggle}>Enter the name for the folder</ModalHeader>
                    <ModalBody>
                        <FormGroup row>
                            <Label for="folderName" sm={3}>Enter the name</Label>
                            <Col sm={9}>
                                <Input type="text" name="folderName" id="folderName" value={name}
                                       onChange={this.changeName} required autoFocus>
                                </Input>
                            </Col>
                        </FormGroup>
                        <div className="clearfix">
                            <Button type="submit" color="success" className="float-right" disabled={disableForm}><i
                                className="fa fa-check fa-lg"/>Create folder</Button>
                        </div>
                        {/*<Input type={"text"} value={name} onChange={this.changeName}*/}
                        {/*       ref={(input) => this.NameInput = input}/>*/}
                    </ModalBody>
                    <ModalFooter>

                    </ModalFooter>
                </Form>
            </Modal>

        );
    }

}

const propTypes = {
    isVisible: PropTypes.bool.isRequired,
    closeModal: PropTypes.func.isRequired,
    containerID: PropTypes.string.isRequired,
    getFilesForContainerID: PropTypes.func.isRequired
};


NewFolder.propTypes = propTypes;


const mapStateToProps = (state, ownProps) => {

    const currentPath = state.explorer.currentPaths[ownProps.containerID];
    let fsInfo = {};

    if (currentPath && state.remote.configs && state.remote.configs[currentPath.remoteName]) {
        fsInfo = state.remote.configs[currentPath.remoteName];
    }
    return {
        currentPath,
        fsInfo
    }
};

export default connect(mapStateToProps, {getFilesForContainerID})(NewFolder);