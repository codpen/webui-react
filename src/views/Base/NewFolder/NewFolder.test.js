import React from "react";
import {shallow} from "enzyme";
import {Provider} from "react-redux";
import {testStore} from "../../../../Utils";
import NewFolder from "./NewFolder";

const setUp = (props = {}) => {
    const component = shallow(
        <Provider store={testStore()}>
            <NewFolder/>
        </Provider>
    );
    return component;
}

describe('New Folder', function () {

    let wrapper;
    beforeEach(() => {
        const props = {};
        wrapper = setUp(props)
    });

    it('should render without crashing', function () {
        expect(wrapper).toHaveLength(1)
    });
});