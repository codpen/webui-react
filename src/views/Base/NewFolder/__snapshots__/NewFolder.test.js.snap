// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Folder renders should match snapshot 1`] = `
<Modal
  autoFocus={true}
  backdrop={true}
  backdropTransition={
    Object {
      "mountOnEnter": true,
      "timeout": 150,
    }
  }
  centered={false}
  data-test="newFolderComponent"
  fade={true}
  isOpen={true}
  keyboard={true}
  modalTransition={
    Object {
      "timeout": 300,
    }
  }
  onClosed={[Function]}
  onOpened={[Function]}
  role="dialog"
  toggle={[Function]}
  zIndex={1050}
>
  <Form
    onSubmit={[Function]}
    tag="form"
  >
    <ModalHeader
      charCode={215}
      closeAriaLabel="Close"
      tag="h5"
      toggle={[Function]}
      wrapTag="div"
    >
      Create New folder at 
      mydrive
      : 
    </ModalHeader>
    <ModalBody
      tag="div"
    >
      <FormGroup
        row={true}
        tag="div"
      >
        <Label
          for="folderName"
          sm={5}
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          Enter the name
        </Label>
        <Col
          sm={7}
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Input
            autoFocus={true}
            id="folderName"
            name="folderName"
            onChange={[Function]}
            required={true}
            type="text"
            value=""
          />
        </Col>
      </FormGroup>
      <div
        className="clearfix"
      >
        <Button
          className="float-right"
          color="success"
          disabled={false}
          tag="button"
          type="submit"
        >
          <i
            className="fa fa-check fa-lg"
          />
          Create folder
        </Button>
      </div>
    </ModalBody>
  </Form>
</Modal>
`;
